SOURCEBUILD.md

```diff
@@ -0,0 +1,237 @@
+# Building from Source
```

Comment:
I like the idea of having one file with all the source build information so that one does not have to browse the whole `contributing.md` to find this info.
Other RAPIDS repositories keep this in `contributing.md` though. This might be confusing for rapids developers when switching between repositories.